plugins/utils/which-key.nix

```diff
@@ -55,257 +126,265 @@ with lib;
-          this is mostly relevant for keymaps that start with a native binding
-        '';
+    icons = {
+      breadcrumb = helpers.defaultNullOpts.mkStr "»" "Symbol used in the command line area that shows your active key combo";
+      separator = helpers.defaultNullOpts.mkStr "➜" "Symbol used between a key and it's label";
+      group = helpers.defaultNullOpts.mkStr "+" "Symbol prepended to a group";
+      ellipsis = helpers.defaultNullOpts.mkStr "…" "Symbol used for overflow ";
+      mappings = helpers.defaultNullOpts.mkBool true "Set to false to disable all mapping icons.";
+      rules =
+        helpers.defaultNullOpts.mkNullable (types.either (types.listOf types.attrs) types.bool) [ ]
+          "Icon rules. Set to false to disable all icons.";
+      colors = helpers.defaultNullOpts.mkBool true ''
+        Use the highlights from mini.icons.
+
+        When `false`, it will use `WhichKeyIcon` instead.
+      '';
+
+      keys = helpers.defaultNullOpts.mkNullable types.attrs {
+        Up = " ";
+        Down = " ";
+        Left = " ";
+        Right = " ";
+        C = "󰘴 ";
+        M = "󰘵 ";
+        D = "󰘳 ";
+        S = "󰘶 ";
+        CR = "󰌑 ";
+        Esc = "󱊷 ";
+        ScrollWheelDown = "󱕐 ";
+        ScrollWheelUp = "󱕑 ";
+        NL = "󰌑 ";
+        BS = "󰁮";
+        Space = "󱁐 ";
+        Tab = "󰌒 ";
+        F1 = "󱊫";
+        F2 = "󱊬";
+        F3 = "󱊭";
+        F4 = "󱊮";
+        F5 = "󱊯";
+        F6 = "󱊰";
+        F7 = "󱊱";
+        F8 = "󱊲";
+        F9 = "󱊳";
+        F10 = "󱊴";
+        F11 = "󱊵";
+        F12 = "󱊶";
+      } "Icons used by key format.";
+    };
+
+    show_help = helpers.defaultNullOpts.mkBool true "show a help message in the command line for using WhichKey";
+
+    show_keys = helpers.defaultNullOpts.mkBool true "show the currently pressed key and its label as a message in the command line";
 
     disable = {
-      buftypes = helpers.defaultNullOpts.mkListOf types.str [ ] "Disabled by default for Telescope";
-      filetypes = helpers.defaultNullOpts.mkListOf types.str [ ] "";
+      bt = helpers.defaultNullOpts.mkListOf types.str [ ] "Disabled by default for Telescope";
+      ft = helpers.defaultNullOpts.mkListOf types.str [ ] "";
     };
+
+    debug = helpers.defaultNullOpts.mkBool false "Enable wk.log in the current directory.";
   };
 
-  config =
+  # TODO: configure
+  settingsExample = { };
+
+  extraOptions = {
+    registrations = mkOption {
+      type = with types; attrsOf anything;
+      description = ''
+        This option is deprecated, use `settings.spec` instead.
+
+        Note: the keymap format has changed in v3 of which-key.
+      '';
+      visible = false;
+    };
+
+    window = mkOption {
+      type = with types; attrsOf anything;
+      description = ''
+        This option is deprecated, use `settings.win` instead.
+
+        Note: the window options have changed parameters in v3 of which-key.
+      '';
+      visible = false;
+    };
+  };
```

Comment:
This would benefit from a `TODO: added/remove after` comment similar to our other deprecations. I guess I neglected to do that when I deprecated `registrations`!